File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/CS/Source/Sample/Forms/ProgressForm.cs
//----------------------------------------------------------------------------- 
// <copyright file="ProgressForm.cs" company="KEYENCE">
//	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
// </copyright>
//----------------------------------------------------------------------------- 
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace LJV7_DllSampleAll
{
	#region Enum
	/// <summary>
	/// Processing status
	/// </summary>
	public enum Status
	{
		Communicating = 0,
		Saving,
	};
	#endregion

	public partial class ProgressForm : Form
	{
		#region Field
		/// <summary>
		/// Processing status
		/// </summary>
		private Status _status;
		#endregion

		#region Properety
		/// <summary>
		/// Processing status
		/// </summary>
		public Status Status
		{
			set
			{
				_status = value;
				switch (_status)
				{
					case Status.Communicating:
						_lblStatus.Text = "Communicating";
						break;
					case Status.Saving:
						_lblStatus.Text = "Saving the file";
						break;
					default:
						return;
				}
			}
			get { return _status; }
		}
		#endregion

		#region Constructor
		/// <summary>
		/// Constructor
		/// </summary>
		public ProgressForm()
		{
			InitializeComponent();
			_status = Status.Communicating;
		}
		#endregion
	}
}

File: LJV-7000 Series - Prolifometer/LJV_communication_library1_4_2/LJV_communication_library1_4_2/Sample/VBNET/Source/Forms/ProgressForm.vb
'----------------------------------------------------------------------------- 
' <copyright file="ProgressForm.cs" company="KEYENCE">
'	 Copyright (c) 2013 KEYENCE CORPORATION.  All rights reserved.
' </copyright>
'----------------------------------------------------------------------------- 
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

#Region "Enum"
''' <summary>
''' Processing status
''' </summary>
Public Enum Status
	Communicating = 0
	Saving
End Enum
#End Region

Public Partial Class ProgressForm
	Inherits Form
	#Region "Field"
	''' <summary>
	''' Processing status
	''' </summary>
	Private _status As Status
	#End Region

	#Region "Properety"
	''' <summary>
	''' Processing status
	''' </summary>
	Public Property Status() As Status
		Get
			Return _status
		End Get
		Set
			_status = value
			Select Case _status
				Case Status.Communicating
					_lblStatus.Text = "Communicating"
					Exit Select
				Case Status.Saving
					_lblStatus.Text = "Saving the file"
					Exit Select
				Case Else
					Return
			End Select
		End Set
	End Property
	#End Region

	#Region "Constructor"
	''' <summary>
	''' Constructor
	''' </summary>
	Public Sub New()
		InitializeComponent()
		_status = Status.Communicating
	End Sub
	#End Region
End Class
